rllib/policy/eager_tf_policy.py

```diff
@@ -471,13 +520,21 @@ def compute_actions_from_input_dict(
 
             # Pass lazy (eager) tensor dict to Model as `input_dict`.
             input_dict = self._lazy_tensor_dict(input_dict)
+            input_dict.is_training = False
             # Pack internal state inputs into (separate) list.
             state_batches = [
                 input_dict[k] for k in input_dict.keys() if "state_in" in k[:8]
             ]
 
-            return self._compute_action_helper(input_dict, state_batches, None,
-                                               explore, timestep)
+            # Call the exploration before_compute_actions hook.
+            self.exploration.before_compute_actions(
+                timestep=timestep, explore=explore, tf_sess=self.get_session())
+
+            ret = self._compute_action_helper(input_dict, state_batches, None,
+                                              explore, timestep)
+            # Update our global timestep by the batch size.
+            self.global_timestep += int(tree.flatten(ret[0])[0].shape[0])
```

Comment:
Yeah, we could probably use `len(input_dict)`, given that it's always a SampleBatch. Otherwise, it'll be trickier, hence the flatten. I think flatten is pretty fast if the structure is not too nested, which normally isn't the case (simple action space).